plugins/filter/to_prettytable.py

```diff
@@ -0,0 +1,181 @@
+
+- name: Display a table with custom alignments
+  debug:
+    msg: "{{ users | community.general.to_prettytable(column_alignments={'name': 'center', 'age': 'right', 'role': 'left'}) }}"
+
+- name: Combine multiple options
+  debug:
+    msg: "{{ users | community.general.to_prettytable(
+        column_order=['role', 'name', 'age'],
+        header_names=['Position', 'Full Name', 'Years'],
+        column_alignments={'name': 'center', 'age': 'right', 'role': 'left'}) }}"
+'''
+
+RETURN = '''
+  _value:
+    description: The formatted ASCII table.
+    type: string
+'''
+
+try:
+    import prettytable
+    HAS_PRETTYTABLE = True
+except ImportError:
+    HAS_PRETTYTABLE = False
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils._text import to_text
+from ansible.module_utils.six import string_types
+
+
+def to_prettytable(data, *args, **kwargs):
+    """Convert a list of dictionaries to an ASCII table.
+
+    Args:
+        data: List of dictionaries to format
+        *args: Optional list of column names to specify column order
+        **kwargs: Optional keyword arguments:
+            - column_order: List of column names to specify the order
+            - header_names: List of custom header names
+            - column_alignments: Dict of column alignments (left, center, right)
+
+    Returns:
+        String containing the ASCII table
+    """
+    if not HAS_PRETTYTABLE:
+        raise AnsibleFilterError(
+            'You need to install "prettytable" Python module to use this filter'
+        )
+
+    # Handle empty data
+    if not data:
+        return "++\n++"
+
+    # Helper function for type error messages
+    def type_error(obj, expected):
+        type_name = "string" if isinstance(obj, string_types) else type(obj).__name__
+        return f"Expected {expected}, got a {type_name}"
+
+    # Validate list type
+    if not isinstance(data, list):
+        raise AnsibleFilterError(type_error(data, "a list of dictionaries"))
+
+    # Validate dictionary items
+    if data and not all(isinstance(item, dict) for item in data):
+        invalid_item = next((item for item in data if not isinstance(item, dict)), None)
+        raise AnsibleFilterError(type_error(invalid_item, "all items in the list to be dictionaries"))
+
+    # Handle positional argument column order
+    column_order = kwargs.get('column_order', None)
+    if args and not column_order:
+        column_order = list(args)
+
+    # Create the table and configure it
+    table = prettytable.PrettyTable()
+
+    # Determine field names
+    field_names = column_order or list(data[0].keys())
+
+    # Set headers
+    header_names = kwargs.get('header_names', None)
+    table.field_names = header_names if header_names else field_names
+
+    # Configure alignments
+    _configure_alignments(table, field_names, kwargs.get('column_alignments', {}))
```

Comment:
Same here.

Reply:
Corrected

Reply:
Added additional validation and corresponding tests.